backend/app/modules/snapshots/finalized/core.py

```diff
@@ -7,3 +15,37 @@ def calculate_leftover(octant_rewards: OctantRewardsDTO, total_withdrawals: int)
         - octant_rewards.operational_cost
         - total_withdrawals
     )
+
+
+def get_finalized_project_rewards(
+    project_settings: ProjectSettings,
+    allocations: list[AllocationDTO],
+    all_projects: list[str],
+    matched_rewards: int,
+) -> tuple[list[ProjectAccountFundsDTO], int]:
+    project_rewards_result = get_projects_rewards(
+        project_settings, allocations, all_projects, matched_rewards
+    )
+
+    return [
+        ProjectAccountFundsDTO(
+            address=r.address, amount=r.allocated + r.matched, matched=r.matched
+        )
+        for r in project_rewards_result.rewards
+        if r.allocated > 0
+    ], project_rewards_result.rewards_sum
+
+
+def get_merkle_root(
+    user_rewards: list[AccountFundsDTO], projects_rewards: list[ProjectAccountFundsDTO]
+):
+    rewards = user_rewards + projects_rewards
+    if not rewards:
```

Comment:
(nonblocking; nitpicking)
won't it be more clear?

```
if rewards == 0:
  ...
```

Reply:
it's a list ;)

Reply:
oh, sure - but the same rule can be applied here, generally I'm not fan of such implicit conversions since not rewards can be true for 0, None, [], etc.; 
the condition like: ```if len(rewards) == 0``` is more straightforward; anyway, it's up to u :v: